### PR TITLE
Add developing documentation for @fluvio/client

### PR DESCRIPTION
**Pull request by Ryanmtate**
Adds some helpful guides on developing the `@fluvio/client` under `DEVELOPING.md`

### CODE REVIEW

File: DEVELOPING.md [hunk 1]
```diff
@@ -1,30 +1,99 @@
...
-To compile rust library:
+Building the `@fluvio/native-<platform>` native modules requires the `Rust` and `Node-Bindgen`. 
+
+Follow the instructions in the links above to install the development tools.
+
+## Building and Testing
+
+### Developing the Native Module
+
+Running `make build` will build the `./src/**/*.{ts,rs}` files, generating a `dist/` folder for the JavaScript files and a native module at the location `./native/src/<platform>/index.node`.
```
Comment (sehz) [inline DEVELOPING.md:28]: This should be only required for non supported platform right?

Comment (Ryanmtate) [reply DEVELOPING.md:28]: This would be required for developing the native module for any platform.

**Review by sehz — APPROVED** [review]

**Comment by sehz** [comment]
@ajhunyady to provide feedback on Node.js API later.

**Comment by ajhunyady** [comment]
@sehz, I ran "Hello World" tutorial in node and it looks good.

**Comment by Ryanmtate** [comment]
> @sehz, I ran "Hello World" tutorial in node and it looks good.

I just ran into an install error, so let me publish a new version.

**Comment by sehz** [comment]
should add  node version 12 to smoke test?

**Comment by Ryanmtate** [comment]
> should add node version 12 to smoke test?

commit 1af31fb re-adds

**Pull request merged**